File: fix-components.js
import fs from 'fs';
import path from 'path';
import { glob } from 'glob';

async function fixComponents() {
  console.log('Fixing component files...');
  
  // Find all component files
  const files = await glob('app/components/**/*.tsx');
  
  let modified = 0;
  
  for (const file of files) {
    try {
      let content = fs.readFileSync(file, 'utf8');
      let originalContent = content;
      
      // Skip empty files
      if (content.trim().length === 0) {
        console.log(`Skipping empty file: ${file}`);
        continue;
      }
      
      // Fix missing interface properties
      if (content.includes('interface') && content.includes('Props')) {
        // Add missing properties to interfaces
        content = content.replace(
          /interface (\w+Props) \{\s*\}/g,
          'interface $1 {\n  className?: string;\n  children?: React.ReactNode;\n}'
        );
      }
      
      // Fix duplicate function declarations
      if (content.includes('const ') && content.includes('export default function')) {
        // Remove the const declaration and keep only the export default function
        const lines = content.split('\n');
        const newLines = [];
        let inConstDeclaration = false;
        let braceCount = 0;
        
        for (let i = 0; i < lines.length; i++) {
          const line = lines[i];
          
          if (line.includes('const ') && line.includes(': React.FC')) {
            inConstDeclaration = true;
            braceCount = 0;
            continue;
          }
          
          if (inConstDeclaration) {
            // Count braces to know when the const declaration ends
            braceCount += (line.match(/\{/g) || []).length;
            braceCount -= (line.match(/\}/g) || []).length;
            
            if (braceCount === 0 && line.includes('};')) {
              inConstDeclaration = false;
              continue;
            }
            
            if (inConstDeclaration) {
              continue;
            }
          }
          
          newLines.push(line);
        }
        
        content = newLines.join('\n');
      }
      
      // Fix stray closing braces
      content = content.replace(/^\s*\}\s*$/gm, '');
      
      // Ensure proper React import
      if (!content.includes('import React')) {
        content = 'import React from \'react\'\n\n' + content;
      }
      
      // Clean up extra empty lines
      content = content.replace(/\n\s*\n\s*\n/g, '\n\n');
      
      if (content !== originalContent) {
        fs.writeFileSync(file, content);
        console.log(`Fixed: ${file}`);
        modified++;
      }
      
    } catch (error) {
      console.error(`Error processing ${file}:`, error.message);
    }
  }
  
  console.log(`Fixed ${modified} component files`);
}

fixComponents().catch(console.error);

File: fix_components.py
#!/usr/bin/env python3
"""
Script to fix malformed component files by creating proper React components
"""
import os
import re
import glob

def fix_component_file(file_path):
    """Fix a single component file by creating a proper React component"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Extract the component name from the file path
        component_name = os.path.basename(file_path).replace('.tsx', '').replace('.ts', '')
        
        # Create a proper React component
        proper_component = f'''import React from 'react';

interface {component_name}Props {{
  // Add props as needed
}}

const {component_name}: React.FC<{component_name}Props> = ({{ ...props }}) => {{
  return (
    <div className="{component_name.lower()}">
      <h2>{component_name}</h2>
      <p>Component content coming soon.</p>
    </div>
  );
}};

export default {component_name};'''
        
        # Write the cleaned content back
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(proper_component)
        
        print(f"Fixed component file: {file_path}")
        return True
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to fix all component files"""
    # Find all component files
    patterns = [
        'app/components/*.tsx',
        'app/components/*.ts'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            # Skip node_modules and other directories
            if 'node_modules' in file_path or '.git' in file_path:
                continue
                
            files_processed += 1
            if fix_component_file(file_path):
                files_fixed += 1
    
    print(f"\nProcessed {files_processed} component files")
    print(f"Fixed {files_fixed} component files")

if __name__ == "__main__":
    main()